src/families/bitcoin/wallet-btc/xpub.ts

```diff
@@ -336,14 +336,44 @@ class Xpub extends EventEmitter {
       needChangeoutput &&
       total.minus(params.amount).minus(fee) > dustAmount
     ) {
-      outputs.push({
-        script: this.crypto.toOutputScript(params.changeAddress.address),
-        value: total.minus(params.amount).minus(fee),
-        address: params.changeAddress.address,
-        isChange: true,
-      });
+      if (total.minus(params.amount).minus(fee).lte(this.OUTPUT_VALUE_MAX)) {
+        outputs.push({
+          script: this.crypto.toOutputScript(params.changeAddress.address),
+          value: total.minus(params.amount).minus(fee),
+          address: params.changeAddress.address,
+          isChange: true,
+        });
+      } else {
+        // change output is too huge and we should split it
+        // fees should be higher since we have multiple change output now
+        const fixedFees =
```

Comment:
Yes but not in this context.
There is still some thing to fix when the amount is too huge because the recipient output will be split into multiple ones and we underestimate the tx size as you mentioned.
but this problem is not directly in the scope of the bug. This bug occurs when the change output (not the recipient change) is too huge and we need to split the change output. It is a blocker of transaction.
The fees problem is not a blocker and I think we can handle it in another PR later.

Reply:
:+1: